src/java/voldemort/store/InsufficientOperationalNodesException.java

```diff
@@ -35,56 +33,45 @@
 
     private static final long serialVersionUID = 1L;
 
-    private Collection<? extends Throwable> causes;
-
     private static final Logger logger = Logger.getLogger(InsufficientOperationalNodesException.class);
 
     public InsufficientOperationalNodesException(String s, Throwable e) {
         super(s, e);
-        causes = Collections.singleton(e);
     }
 
     public InsufficientOperationalNodesException(String s) {
         super(s);
-        causes = Collections.emptyList();
     }
 
     public InsufficientOperationalNodesException(Throwable e) {
         super(e);
-        causes = Collections.singleton(e);
     }
 
-    public InsufficientOperationalNodesException(Collection<? extends Throwable> failures) {
+    public InsufficientOperationalNodesException(List<? extends Throwable> failures) {
         this("Insufficient operational nodes to immediately satisfy request.", failures);
     }
 
-    public InsufficientOperationalNodesException(String message,
-                                                 Collection<? extends Throwable> failures) {
-        super(message, failures.size() > 0 ? failures.iterator().next() : null);
-        this.causes = failures;
+    public InsufficientOperationalNodesException(String message, List<? extends Throwable> failures) {
+        super(message, failures.size() > 0 ? failures.get(0) : null);
     }
 
     public InsufficientOperationalNodesException(String message,
                                                  List<Node> replicationSet,
                                                  List<Node> preferenceList,
                                                  List<Node> failedList,
-                                                 Collection<? extends Throwable> failures) {
+                                                 List<? extends Throwable> failures) {
         this(message + " Original replication set :" + stripNodeIds(replicationSet)
                      + " Known failed nodes before operation :"
                      + stripNodeIds(difference(replicationSet, preferenceList))
                      + " Estimated live nodes in preference list :" + stripNodeIds(preferenceList)
                      + " New failed nodes during operation :"
                      + stripNodeIds(difference(failedList, replicationSet)),
-             failures.size() > 0 ? failures.iterator().next() : null);
+             failures.size() > 0 ? failures.get(0) : null);
```

Comment:
Why do we even provide a constructor that accepts a list of failures if we're going to cherry pick the first one and discard the rest?

I see that you're deleting the unused `getCauses()` function. That's fine if we don't use it, but if we're going to clean up this class, then I think we should also clean up the constructor. It's a bit misleading the way it is now...